### ISSUE TITLE
Add quick utils functions to get B and C matrices from off-resonance maps

### ISSUE THREAD

**Issue by chaithyagr** [mind-inria/mri-nufft#124]
Currently, in

https://github.com/mind-inria/mri-nufft/blob/f5a6ba006184b96a1f73ff54be27d854797c79fe/src/mrinufft/operators/off_resonnance.py#L19-L58

we just generate the fourier_op which takes the B and C values into account. 

- [ ] It might help to also include a function in `mrinufft.operators.off_resonance` called `get_interpolators_from_fieldmap`.
- [ ] Also, it might further help to be able to get fourier correction possible directly from the __init__ function itself like `MRIFourierCorrected(fourier_op, b0_field, mask)`
- [ ] It could be useful to allow for autodiff to pass through this new fourier op, currently we only support cupy or numpy, it might help to check if we have autograd enabled on based fourier_op and enable it as well (we just need to have tensors as torch tensors.
- [ ] A bit distant, but it could help to have the off-resonance estimation from k-space data as function. Although, this is a bit distant and I really feel it should belong to pysap-mri.

**Comment by paquiteau** [comment]
I fully agree with you. 

Getting the off resonance map from the k space data could be done in the extras module, just as we propose it  for the smaps estimation. 

We may also wish to  have autodiff support around this as well ?

**Comment by chaithyagr** [comment]
Sure. But note that the estimation isnt trivial, it is computationally expensive (for now). @Daval-G you think these codes can be open sourced though? I wanted to confirm that.

**Comment by philouc** [comment]
I agree with @paquiteau : Estimation of off-resonance effects from k-space data should be carried out separately, as it depends on the phase information and TE value usually extracted from the image domain.
